### ISSUE TITLE
Duplicate PeekDefinition results for TS IGoToDefinitionService implementation

### ISSUE THREAD

**Issue by dibarbet** [dotnet/roslyn#38483]
https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS?path=%2FVS%2FLanguageService%2FTypeScriptLanguageService%2FShared%2FFeatures%2FGoToDefinition%2FTypeScriptGoToDefinitionService.cs&version=GBmaster&line=39&lineStyle=plain&lineEnd=40&lineStartColumn=1&lineEndColumn=1

TS provides their own peek definition handling.  Roslyn also calls the IGoToDefinitionService for peek definition, which results in duplicate results for TS when `FindDefinitionsAsync` is implemented on their side.

This causes issues in liveshare, because now we have to use FAR for TS go to definition requests.

We need to do something to resolve the duplicate results for peek definition (flag, prevent roslyn from handling it, etc.) so we can re-implement IGoToDefinitionService for TS and use it in liveshare.

**Comment by CyrusNajmabadi** [comment]
> TS provides their own peek definition handling.

Why?  

The point of Roslyn is to not have to have the individual languages provide editor features.  Instead, roslyn provides the editor integration, and the languages just need to provide the language smarts.

**Comment by amcasey** [comment]
@CyrusNajmabadi It's been a while, but I believe the issue was that we wanted to be able to peek into files without corresponding Documents.  Since our impl of `IPeekableItemSourceProvider` was sufficiently different from C#'s, we exported our own.  That worked until we discovered that Roslyn was still calling the `FindDefinitionsAsync` for Peek.

**Comment by CyrusNajmabadi** [comment]
Hi @amcasey, what's an example of a file you want to peek into that doesn't have a document?  The idea itself seems pretty fishy to me since it is effectively saying that the abstraction we have for something peekable isn't something we want to use here.

**Comment by amcasey** [comment]
@CyrusNajmabadi I think it's conceptually similar to metadata-as-source.  In TS, you can reference another library/project through its definition (aka .d.ts) file.  To make cross-project GTD, FAR, etc work, we added some code to map from a .d.ts file back to the originating .ts file (best effort).  If that file isn't part of a loaded project, we open it as a loose file, since we presume it is primarily being read (vs written).

**Comment by CyrusNajmabadi** [comment]
Right, but with metadata-as-source, even the generated code file is a 'document'.  I would expect the same in the scenario you mentioned.  'loose files' are definitely representable in our world :-) 

Note: if you don't have a document, then how do any language services work in the file you 'peeked' into?

**Comment by amcasey** [comment]
I haven't looked at that code in a while, but I believe we create the document after it's opened.  @minestarks might have more context.

**Comment by CyrusNajmabadi** [comment]
Closing out as speculative.  If this does need to be fixed, assign to a milestone.
